secret/driver/azure.go

```diff
@@ -90,7 +90,23 @@ func (driver *Azure) Encrypt(input string) (string, error) {
 	return base64.RawStdEncoding.EncodeToString(res.Result), nil
 }
 
-func (driver *Azure) Decrypt(input string) (string, error) {
+// Decrypt decrypts an input either using the key configured in the driver or if the key parameter isn't empty it will use that one.
+func (driver *Azure) Decrypt(input string, key string) (string, error) {
+	var err error
+	keyName := driver.config.KeyName
+	keyVersion := driver.config.KeyVersion
+
+	// if we hand over a key to this func, make sure to use this key for the decryption, otherwise use the key configured in the driver
+	if len(key) > 0 {
+		_, keyName, keyVersion, err = parseAzureKeyVaultKeyId(key)
+		if err != nil {
+			return "", fmt.Errorf("the key we handed over to Decrypt() could not be parsed")
+		}
+		if keyName != driver.config.KeyName || keyVersion != driver.config.KeyVersion {
+
+		}
```

Comment:
Here seems to be missing something?

Reply:
oops, that should not have been committed.